File: nodes/array-slice/array-slice.process.py
async def process(inputs, settings, config, nodeConfig):
  array = inputs.get("array", [])
  if not isinstance(array, list):
    array = [array]
    
  start = int(inputs.get("start", 0))
  end = int(inputs.get("end")) if inputs.get("end") is not None else None
  
  return {
    "array": array[start:end]
  } 

File: nodes/array-slice/array-slice.process.js
export default async ({inputs, settings, config}) => {

  const array = Array.isArray(inputs.array) ? inputs.array : [inputs.array];
  const start = inputs.start !== undefined ? parseInt(inputs.start) : 0;
  const end = inputs.end !== undefined ? parseInt(inputs.end) : undefined;
  
  return {
    array: array.slice(start, end)
  };
}; 